### ISSUE TITLE
ufw_appinfo doesn't work with ufw app info all?

### ISSUE THREAD

**Issue by chriscroome** [kellyjonbrazil/jc#394]
The [documentation](https://kellyjonbrazil.github.io/jc/docs/parsers/ufw_appinfo) contains:

> Supports individual apps via `ufw app info [application]` and all apps list via `ufw app info all`.

This works for one app:

```bash
ufw app info OpenSSH | jc --ufw-appinfo -yp
```
```yml
---
- profile: OpenSSH
  title: Secure shell server, an rshd replacement
  description: OpenSSH is a free implementation of the Secure Shell protocol.
  tcp_list:
  - 22
  normalized_tcp_list:
  - 22
```

However the `all` option doesn't appear to work?

```bash
ufw app info all | jc --ufw-appinfo -yp
jc:  Error - ufw-appinfo parser could not parse the input data.
             If this is the correct parser, try setting the locale to C (LC_ALL=C).
             For details use the -d or -dd option. Use "jc -h --ufw-appinfo" for help.
```
```bash
LC_ALL=C ufw app info all | jc --ufw-appinfo -yp
jc:  Error - ufw-appinfo parser could not parse the input data.
             If this is the correct parser, try setting the locale to C (LC_ALL=C).
             For details use the -d or -dd option. Use "jc -h --ufw-appinfo" for help.
```
```bash
export LANG=C && export LC_NUMERIC=C && ufw app info all | jc --ufw-appinfo -yp
jc:  Error - ufw-appinfo parser could not parse the input data.
             If this is the correct parser, try setting the locale to C (LC_ALL=C).
             For details use the -d or -dd option. Use "jc -h --ufw-appinfo" for help.
```

The command itself seems fine:

```bash
ufw app info all | head -n 18
Profile: AIM
Title: AIM Talk
Description: AIM talk protocol

Port:
  5190/tcp

--

Profile: Bonjour
Title: Bonjour
Description: Bonjour protocol

Ports:
  5353/udp
  5298

--
```

However I must admit that I prefer using the output the INI parser generates for this, for example:

```bash
cat /etc/ufw/applications.d/* | jc --ini -yp | tail -n 8
```
```yml
WWW Full:
  title: Web Server (HTTP,HTTPS)
  description: Web Server (HTTP,HTTPS)
  ports: 80,443/tcp
WWW Cache:
  title: Web Server (8080)
  description: Web Server (8080)
  ports: 8080/tcp
```

So I have no need for it to be fixed...

**Comment by kellyjonbrazil** [comment]
Hi - thanks for reporting this. The parser seems to work fine on the small snippet of output that you provided. Could you send over the whole output of `ufw app info all` and I'll see what's going on? Thanks!

**Comment by chriscroome** [comment]
Following is what I have on the server where I found the error, this server does have files I added myself (I can provide these if that helps), however on a Debian machine where I haven't added any additional applications I also have the same error.

```bash
ufw app info all
```
```
Profile: AIM
Title: AIM Talk
Description: AIM talk protocol

Port:
  5190/tcp

--

Profile: Bonjour
Title: Bonjour
Description: Bonjour protocol

Ports:
  5353/udp
  5298

--

Profile: CIFS
Title: SMB/CIFS server
Description: SMB/CIFS server

Ports:
  137,138/udp
  139,445/tcp

--

Profile: DNS
Title: Internet Domain Name Server
Description: Internet Domain Name Server

Port:
  53

--

Profile: Deluge
Title: Deluge
Description: Deluge BitTorrent client

Port:
  6881:6891/tcp

--

Profile: IMAP
Title: Mail server (IMAP)
Description: Mail server (IMAP)

Port:
  143/tcp

--

Profile: IMAPS
Title: Secure mail server (IMAPS)
Description: Secure mail server (IMAPS)

Port:
  993/tcp

--

Profile: IPP
Title: Cups server (IPP)
Description: Cups server (IPP)

Port:
  631

--

Profile: Icinga
Title: Icinga
Description: Icinga monitoring

Port:
  5665/tcp

--

Profile: KTorrent
Title: KTorrent
Description: KTorrent BitTorrent client

Ports:
  6881/tcp
  4444/udp

--

Profile: Kerberos Admin
Title: Kerberos v5 admin
Description: Kerberos v5 server

Port:
  749/tcp

--

Profile: Kerberos Full
Title: Kerberos v5 server
Description: Kerberos v5 server

Ports:
  88,749/tcp
  464/udp

--

Profile: Kerberos KDC
Title: Kerberos v5 KDC server
Description: Kerberos v5 KDC server

Port:
  88

--

Profile: Kerberos Password
Title: Kerberos v5 password
Description: Kerberos v5 password

Port:
  464/udp

--

Profile: LDAP
Title: LDAP server
Description: LDAP server

Port:
  389/tcp

--

Profile: LDAPS
Title: LDAP server (LDAPS)
Description: LDAP server (LDAPS)

Port:
  636/tcp

--

Profile: LPD
Title: LPD server
Description: LPD server

Port:
  515/tcp

--

Profile: MSN
Title: MSN Chat
Description: MSN chat protocol (with file transfer and voice)

Ports:
  1863
  6891:6900/tcp
  6901

--

Profile: MSN SSL
Title: MSN Chat (SSL)
Description: MSN chat protocol (SSL)

Port:
  443/tcp

--

Profile: Mail submission
Title: Mail server (Submission)
Description: Mail server (Submission)

Port:
  587/tcp

--

Profile: Munin
Title: Munin
Description: Munin monitoring

Port:
  4949/tcp

--

Profile: NFS
Title: NFS server
Description: NFS and portmap server. Will also need access to mountd, statd
and possibly others

Ports:
  2049,111/tcp
  2049,111/udp

--

Profile: Nginx Full
Title: Web Server (Nginx, HTTP + HTTPS)
Description: Small, but very powerful and efficient web server

Ports:
  80,443/tcp

--

Profile: Nginx HTTP
Title: Web Server (Nginx, HTTP)
Description: Small, but very powerful and efficient web server

Port:
  80/tcp

--

Profile: Nginx HTTPS
Title: Web Server (Nginx, HTTPS)
Description: Small, but very powerful and efficient web server

Port:
  443/tcp

--

Profile: OpenSSH
Title: Secure shell server, an rshd replacement
Description: OpenSSH is a free implementation of the Secure Shell protocol.

Port:
  22/tcp

--

Profile: POP3
Title: Mail server (POP3)
Description: Mail server (POP3)

Port:
  110/tcp

--

Profile: POP3S
Title: Secure mail server (POP3S)
Description: Secure mail server (POP3S)

Port:
  995/tcp

--

Profile: PeopleNearby
Title: People Nearby
Description: People Nearby (Bonjour/Salut) functionality in Empathy

Ports:
  5353/udp
  5298

--

Profile: RabbitMQ
Title: RabbitMQ
Description: RabbitMQ is the most widely deployed open source message
broker.

Ports:
  4369/tcp
  5671:5672/tcp
  5551:5552/tcp
  6000:6500/tcp
  25672/tcp
  35672:35682/tcp

--

Profile: SMTP
Title: Mail server (SMTP)
Description: Mail server (SMTP)

Port:
  25/tcp

--

Profile: SSH
Title: SSH server
Description: SSH server

Port:
  22/tcp

--

Profile: Socks
Title: Socks proxy
Description: Socks proxy

Port:
  1080/tcp

--

Profile: Telnet
Title: Telnet server (insecure)
Description: Telnet server (insecure)

Port:
  23/tcp

--

Profile: Transmission
Title: Transmission
Description: Transmission BitTorrent client

Port:
  51413

--

Profile: Transparent Proxy
Title: Transparent proxy
Description: Transparent proxy

Port:
  8081/tcp

--

Profile: Turnserver
Title: Coturn Turnserver
Description: Free open source implementation of TURN and STUN Server

Ports:
  3478,3479,5349,5350,49152:65535/tcp
  3478,3479,5349,5350,49152:65535/udp

--

Profile: VNC
Title: VNC server
Description: VNC server

Port:
  5900/tcp

--

Profile: WWW
Title: Web Server
Description: Web server

Port:
  80/tcp

--

Profile: WWW Cache
Title: Web Server (8080)
Description: Web Server (8080)

Port:
  8080/tcp

--

Profile: WWW Full
Title: Web Server (HTTP,HTTPS)
Description: Web Server (HTTP,HTTPS)

Ports:
  80,443/tcp

--

Profile: WWW Secure
Title: Web Server (HTTPS)
Description: Web Server (HTTPS)

Port:
  443/tcp

--

Profile: XMPP
Title: XMPP Chat
Description: XMPP protocol (Jabber and Google Talk)

Ports:
  5222/tcp
  5269/tcp

--

Profile: Yahoo
Title: Yahoo Chat
Description: Yahoo chat protocol

Port:
  5050

--

Profile: qBittorrent
Title: qBittorrent
Description: qBittorrent BitTorrent client

Port:
  6881/tcp

--

Profile: svnserve
Title: Subversion server
Description: Subversion server for access to Subversion repositories.

Port:
  3690/tcp
```

**Comment by kellyjonbrazil** [comment]
Thanks - it looks like it works with the `-r` (raw) option and it's failing in the `_process` function. I'll check it out.

**Comment by kellyjonbrazil** [comment]
This one looks like it's the culprit. The Description spans more than one line, which is causing the parsing issue:
```
Profile: RabbitMQ
Title: RabbitMQ
Description: RabbitMQ is the most widely deployed open source message
broker.

Ports:
  4369/tcp
  5671:5672/tcp
  5551:5552/tcp
  6000:6500/tcp
  25672/tcp
  35672:35682/tcp
```
I'll take a closer look. Hopefully a quick fix.

**Comment by kellyjonbrazil** [comment]
I have a fix for this in the `dev` branch:
https://github.com/kellyjonbrazil/jc/blob/dev/jc/parsers/ufw_appinfo.py

You can test by copying the file above to the parser plugin folder:
- Linux/unix: `$HOME/.local/share/jc/jcparsers`
- macOS: `$HOME/Library/Application Support/jc/jcparsers`
- Windows: `$LOCALAPPDATA\jc\jc\jcparsers`

Let me know if you run into any other issues. Thanks!

**Comment by kellyjonbrazil** [comment]
Released in v1.23.2
